lib/application_insights/channel/contracts/data.rb

```diff
@@ -1,48 +1,9 @@
 require_relative 'json_serializable'
 
-module ApplicationInsights
-  module Channel
-    module Contracts
-      # Data contract class for type Data.
-      class Data < JsonSerializable
-        # Initializes a new instance of the Data class.
-        def initialize(options={})
-          defaults = {
-            'baseType' => nil,
-            'baseData' => nil
-          }
-          values = {
-            'baseData' => nil
-          }
-          super defaults, values, options
-        end
-        
-        # Gets the base_type property.
-        def base_type
-          @values.fetch('baseType') { 
-            @values['baseType'] = nil
-          }
-        end
-        
-        # Sets the base_type property.
-        def base_type=(value)
-          if value == @defaults['baseType']
-            @values.delete 'baseType' if @values.key? 'baseType'
-          else
-            @values['baseType'] = value
-          end
-        end
-        
-        # Gets the base_data property.
-        def base_data
-          @values['baseData']
-        end
-        
-        # Sets the base_data property.
-        def base_data=(value)
-          @values['baseData'] = value
-        end
-      end
-    end
+module ApplicationInsights::Channel::Contracts
+  class Data
+    include JsonSerializable
+
+    attributes :baseType, :baseData
```

Comment:
Since the contract classes are generated, I'd like to have the decorator inside the generator code (which uses different decorate patterns to generate the ruby style compliant code).  There is a plan to open sourcing the generator, hopefully in the near future.

For example 1 and example 2, it seems not necessary to introduce the new method 'attributes' but using attr_accessor, attr_writer directly, and maintain the mapping internally to server the serialization purpose only. Such as: 

class MessageData < JsonSerializable

  def initialize(options={})
    mapping = {
        'ver' => 'ver',
        'message' => 'message',
        'severity_level' => 'severityLevel',
        'properties' => 'properties'
    }
    super mapping, options
  end

  # For fields with default value, write attribute only and define the read method with default value.
  attr_writer :ver, :properties
  attr_accessor :message, :severity_level

  def ver
    @ver ||= 2
  end

  def properties
    @properties ||= {}
  end

end